### DISCUSSION TITLE
berry timer remaining term

### DISCUSSION THREAD

**Discussion by DetlefR** [arendst/Tasmota#20874] [Q&A]
Hello,

in a berry script I use a timer. And I want to reset this timer with different values. But only if the remaining term shorter than the new one.
Therefore, I must know the remaining term for the timer. Is there a way to find the remaining term.

Thanks
Detlef

**Comment by barbudor** [comment]
AFAIK there is no way to know the current status or update a timer
The only possibility, if you have given an ID on creation is to delete it

@s-hadinger It could be nice to get a timer object returned by `tasmota.set_timer()` in order to interrogate, update
Or similar function as removed, based on ID

**Reply by sfromis** [reply]
Current implementation of the timer includes no "remaining time"; instead checking if the calculated end time is reached, at short intervals. Due to the risk of sign and overflow issues, calculating remaining time would be a bit more complex than just subtracting.

**Reply by s-hadinger** [reply]
Timers are used extensively. I'm reluctant to consume more memory for a rarely used feature. For this kind of usage I would suggest to do your own implementation. 

Currently it's just a timestamp in millis and a closure.

**Reply by s-hadinger** [reply]
In addition, there is no natural identifier for a timer, so this would need more memory to store them

**Reply by sfromis** [reply]
Well, there's already an `id` field in the `Trigger` object and `tasmota.set_timer`, at least.

**Reply by s-hadinger** [reply]
Oh right. Scratch what I said :D

**Reply by DetlefR** [reply]
Too bad, I'll then solve this using a variable with the calculated time.

Thanks
